src/format/parsed.rs

```diff
@@ -514,10 +515,17 @@ impl Parsed {
         };
         let hour = hour_div_12 * 12 + hour_mod_12;
 
+        // we allow omitting minutes, but only if seconds and nanoseconds are also missing.
         let minute = match self.minute {
             Some(v @ 0..=59) => v,
             Some(_) => return Err(OUT_OF_RANGE),
-            None => return Err(NOT_ENOUGH),
+            None => {
+                if self.second.is_none() && self.nanosecond.is_none() {
+                    0
+                } else {
+                    return Err(NOT_ENOUGH);
+                }
+            }
```

Comment:
Style: writing this as a `match` expression will be slightly more concise and IMO easier to read.